blackbox/docs/general/builtins/window-functions.rst

```diff
@@ -219,6 +228,102 @@ Example::
    +---------+-----+---------------------------+
    SELECT 18 rows in set (... sec)
 
+.. _named-windows:
+
+Named windows
+-------------
+
+It is possible to define a list of named window definitions that can be
+referenced in :ref:`over` clauses. To do this, use the
+:ref:`sql_reference_window` clause in the :ref:`sql_reference_select` clause.
+
+Named windows are particularly useful when the same window definition
+could be used in multiple :ref:`over` clauses. For instance
+
+::
+
+   cr> SELECT
+   ...   x,
+   ...   FIRST_VALUE(x) OVER (w) AS "first",
+   ...   LAST_VALUE(x) OVER (w) AS "last"
+   ... FROM unnest([1, 2, 3, 4]) as t(x)
+   ... WINDOW w AS (ORDER BY x)
+   +---+-------+------+
+   | x | first | last |
+   +---+-------+------+
+   | 1 |     1 |    1 |
+   | 2 |     1 |    2 |
+   | 3 |     1 |    3 |
+   | 4 |     1 |    4 |
+   +---+-------+------+
+   SELECT 4 rows in set (... sec)
+
+If a ``window_name`` is specified in the window definition of the :ref:`over`
+clause, then there must be a named window entry that matches the ``window_name``
+in the window definition list of the :ref:`sql_reference_window` clause.
+
+If the :ref:`over` clause has its own non-empty window definition and
+references a window definition from the :ref:`sql_reference_window` clause,
+then it can only add clauses from the referenced window, but not overwrite them.
+
+::
+
+   cr> SELECT
+   ...   x,
+   ...   LAST_VALUE(x) OVER (w ORDER BY x)
+   ... FROM unnest(
+   ...      [1, 2, 3, 4],
+   ...      [1, 1, 2, 2]) as t(x, y)
+   ... WINDOW w AS (PARTITION BY y)
+   +---+---+
+   | x | y |
+   +---+---+
+   | 1 | 1 |
+   | 2 | 2 |
+   | 3 | 3 |
+   | 4 | 4 |
+   +---+---+
+   SELECT 4 rows in set (... sec)
+
+Otherwise, an attempt to override the clauses of the referenced window
+by the window definition of the :ref:`OVER` clause will result in failure.
+
+::
+
+   cr> SELECT
+   ...   FIRST_VALUE(x) OVER (w ORDER BY x)
+   ... FROM unnest([1, 2, 3, 4]) as t(x)
+   ... WINDOW w AS (ORDER BY x)
+   SQLActionException[SQLParseException: Cannot override ORDER BY clause of window w]
+
+It is not possible to define the ``PARTITION BY`` clause in the window
+definition of the :ref:`OVER` clause if it references a window definition
+from the ref:`sql_reference_window` clause.
+
+The window definitions in the :ref:`sql_reference_window` clause cannot define
+its own window frames, if they are referenced by non-empty window definitions
+of the :ref:`OVER` clauses.
+
+The definition of the referenced window can itself begin with a ``window_name``.
```

Comment:
I think this could use some explanation on what's happening in this case. From the example alone it is not obvious.